src/mavlink/manager.rs

```diff
@@ -1 +1,203 @@
+}
+
+pub struct Manager {
+    connection: Arc<Mutex<Connection>>,
+    ids: Arc<Mutex<Vec<u8>>>,
+}
+
+struct Connection {
+    address: String,
+    connection: Box<dyn MavConnection<MavMessage> + Sync + Send>,
+    sender: broadcast::Sender<Message>,
+}
+
+#[derive(Debug, Clone)]
+pub enum Message {
+    Received((MavHeader, MavMessage)),
+    ToBeSent((MavHeader, MavMessage)),
+}
+
+impl Default for Manager {
+    #[instrument(level = "debug")]
+    fn default() -> Self {
+        let address =
+            settings::manager::mavlink_endpoint().expect("No configured mavlink endpoint");
+
+        let connection = Connection::connect(&address);
+
+        let (sender, _receiver) = broadcast::channel(100);
+
+        let this = Self {
+            connection: Arc::new(Mutex::new(Connection {
+                address,
+                connection,
+                sender,
+            })),
+            ids: Arc::new(Mutex::new(vec![])),
+        };
+
+        let connection = this.connection.clone();
+        std::thread::Builder::new()
+            .name("MavSender".into())
+            .spawn(move || Manager::sender_loop(connection))
+            .expect("Failed to spawn MavSender thread");
+
+        let connection = this.connection.clone();
+        std::thread::Builder::new()
+            .name("MavReceiver".into())
+            .spawn(move || Manager::receiver_loop(connection))
+            .expect("Failed to spawn MavReceiver thread");
+
+        this
+    }
+}
+
+impl Manager {
+    // Construct our manager, should be done inside main
+    #[instrument(level = "debug")]
+    pub fn init() {
+        MANAGER.as_ref();
+    }
+
+    #[instrument(level = "debug", skip(inner))]
+    fn receiver_loop(inner: Arc<Mutex<Connection>>) {
+        loop {
+            std::thread::sleep(std::time::Duration::from_micros(10));
+
+            // Receive from the Mavlink network
+            let (header, message) = match inner.lock().unwrap().connection.recv() {
+                Ok(message) => message,
+                Err(error) => {
+                    if let mavlink::error::MessageReadError::Io(io_error) = &error {
+                        if io_error.kind() == std::io::ErrorKind::WouldBlock {
+                            trace!("Blocked I/O, probably disconnected!");
+                            continue;
+                        }
+                    }
+                    error!("Failed receiving message from Mavlink Connection: {error:?}");
+                    break;
+                }
+            };
+
+            trace!("Message received: {header:?}, {message:?}");
+
+            // Send the received message to the cameras
+            if let Err(error) = inner
+                .lock()
+                .unwrap()
+                .sender
+                .send(Message::Received((header, message)))
+            {
+                error!("Failed handling message: {error:?}");
+                continue;
```

Comment:
this continue is unnecessary.

Reply:
I agree, but if some day we decide to add extra lines below...